mavros_extras/src/plugins/odom.cpp

```diff
@@ -156,17 +161,15 @@ class OdometryPlugin : public plugin::PluginBase {
 		mavlink::common::msg::ATTITUDE_QUATERNION_COV att {};
 
 		att.time_usec = stamp_usec;
-		ftf::quaternion_to_mavlink(q_ned, att.q);
-		ftf::covariance_to_mavlink(cov_vel, att.covariance);
+		ftf::quaternion_to_mavlink(q_parent2ned, att.q);
+		att_cov_ned = tf_child2ned.linear()*twist_cov_child.block<3, 3>(0, 0)*tf_child2ned.linear().transpose();
+		for (int i=0; i<3; i++) for (int j=0; j<3; j++) {
+			att.covariance[i*3 + j] = att_cov_ned(i, j);
+		}
```

Comment:
Maybe `Matrix::array()` is what you need to pass to `ftf::covariance_to_mavlink()`?